ui/components/app/detected-token/detected-token.js

```diff
@@ -43,13 +48,42 @@ const DetectedToken = ({ setShowDetectedTokens }) => {
   ] = useState(false);
 
   const handleClearTokensSelection = async () => {
-    // create a lodash chain on this object
     const {
-      selected: selectedTokens,
-      deselected: deSelectedTokens,
+      selected: selectedTokens = [],
+      deselected: deSelectedTokens = [],
     } = sortingBasedOnTokenSelection(tokensListDetected);
 
+    if (deSelectedTokens.length > 0) {
+      const tokensDetailsList = deSelectedTokens.map(
+        ({ symbol, address }) => `${symbol} - ${address}`,
+      );
+      trackEvent({
+        event: EVENT_NAMES.TOKEN_HIDDEN,
+        category: EVENT.CATEGORIES.WALLET,
+        sensitiveProperties: {
+          tokens: tokensDetailsList,
+          location: EVENT.LOCATION.TOKEN_DETECTION,
+          token_standard: TOKEN_STANDARDS.ERC20,
+          asset_type: ASSET_TYPES.TOKEN,
+        },
+      });
+    }
+
     if (deSelectedTokens.length < detectedTokens.length) {
+      selectedTokens.forEach((importedToken) => {
```

Comment:
NIT: I think we can improve the readability of this code by extracting the add token logic into its own function

```
      // const selectedTokens = ... 
      
      const tokenSymbols = selectedTokens.map(({ symbol }) => symbol);
      await dispatch(importTokens(selectedTokens));
      dispatch(setNewTokensImported(tokenSymbols.join(', ')));

      // trackEvent logic
```